recipes/bdwgc/all/patches/update-cmake-8_0_6.patch

```diff
@@ -323,11 +323,9 @@
 +  endif()
 +  include_directories(libatomic_ops/src)
 +  if (MSVC)
-+    # TODO: Rename the following to Atomic_ops
-+    find_package(libatomic_ops CONFIG)
-+    if (libatomic_ops_FOUND)
-+      # TODO: Rename the following to Atomic_ops::atomic_ops
-+      get_target_property(AO_INCLUDE_DIRS libatomic_ops::libatomic_ops
++    find_package(Atomic_ops CONFIG)
++    if (Atomic_ops_FOUND)
++      get_target_property(AO_INCLUDE_DIRS Atomic_ops::atomic_ops
 +                          INTERFACE_INCLUDE_DIRECTORIES)
 +      include_directories(${AO_INCLUDE_DIRS})
```

Comment:
It is probably preferable to use target-based functions to propagate this. Would it be possible to link the necessary target against `libatomic_ops` using `target_link_libraries`? A generator expression can be used to only link the target if it exists, assuming CMake 3.12 is available:

```cmake
target_link_libraries(<target> PRIVATE $<TARGET_NAME_IF_EXISTS:Atomic_ops::atomic_ops>)
```

Reply:
Right in general, this is done in the upstream master - https://github.com/ivmai/bdwgc/blob/master/CMakeLists.txt#L544
But, in our case (libatomic_ops is needed only for MS VC compiler and it is header-only library in case of MS VC) it is not needed for a successful build. Again, I've followed the principle "the short patch the better".

Reply:
Thanks for the response! That makes sense.